marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubInputPartitionReader.java

```diff
@@ -117,4 +122,19 @@ private void readNextBatchOfRows() {
             rowIndex = 0;
         }
     }
+
+    private ObjectNode addOptimizationLevel(Map<String, String> options, ObjectNode endpointConstants){
+        String optimizationlevel = options.get("optimizationlevel");
+        if(optimizationlevel!=null && optimizationlevel.length()>0) {
```

Comment:
It's best to use StringUtils.hasText or StringUtils.isNotEmpty here, as that will do a trim().length(), which rules out whitespace.